lib/ansible/module_utils/basic.py

```diff
@@ -624,6 +629,10 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         if not self.no_log and self._verbosity >= 3:
             self._log_invocation()
 
+        if not self.no_log and self._debug and self._remote_log_target:
+            self.debug_logger = logging.getLogger(self.__class__.__name__)
```

Comment:
I'd like a shorter name[1] here, but it looks like self.log and self.debug are taken.
self.dl is likely too far.

[1] since there will be a lot of ```self.debug_logger.debug('foo %s %s', obj1, otherobj)```
and ```self.debug_logger.exception(e)```